### PR TITLE
wayfreeze: 0-unstable-2024-05-23 -> 0-unstable-2024-09-20

### PR DESCRIPTION
**Pull request by purrpurrn**
## Description of changes

Updated it to the latest commit and added r-ryantm auto-updating

Changelogs:
- fix: fractional scaling
- fix: typos
- fix: problems with buffer
- fix: crash if the user uses sway
- feature: new args

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Jappie3 — APPROVED** [review]
this means I should **really** stop using master as a testing ground huh? thanks lmao ^^

File: pkgs/by-name/wa/wayfreeze/package.nix [hunk 1]
```diff
@@ -2,20 +2,23 @@
...
   src = fetchFromGitHub {
     owner = "Jappie3";
     repo = "wayfreeze";
-    rev = "069dea0b832bd5b7a7872a57bd53f51cd377f206";
-    hash = "sha256-3btFzZbkHT6kBBA3M7OwFsD710VpMiHSXIpHmvCD/es=";
+    rev = "dcbe2690ce41a286ef1eed54747bac47cee6dc2c";
+    hash = "sha256-XlZSVN/kTSA5X/kTpD/Hr5YBXdfh8gJPq5Da4tL0Gpk=";
   };
 
+  passthru.updateScript = unstableGitUpdater { };
```
[the remark this answers is not in the document]
Comment (Jappie3) [reply pkgs/by-name/wa/wayfreeze/package.nix:20]: grepping through nixpkgs, I can find several occurrences of `nix-update-script { extraArgs = [ "--version=branch" ]; };`, so I don't think you should specify a branch name

**Comment by Atemu** [comment]
> this means I should **really** stop using master as a testing ground huh? thanks lmao ^^

You could also tag releases instead. We only track master when no viable released version is available.

**Comment by Jappie3** [comment]
I'll look into it, thanks

**Review by Atemu — CHANGES_REQUESTED** [review]

**Review by Atemu — APPROVED** [review]
Diff LGTM.

Please rebase atop of master.

**Review by Scrumplex — APPROVED** [review]

**Review by Jappie3 — APPROVED** [review]

**Pull request merged**